src/HAEntity.cpp

```diff
@@ -72,6 +73,22 @@ std::string HAEntity::getNameFromState(void)
   }
 }
 
+void HAEntity::attach(IObserver* observer)
+{
+  uientities.insert(observer);
```

Comment:
maybe `uientities` should just be `observers` here? The pattern is nice and generic, and who knows what else might care in the future. Just adding this suggestion as a comment is fine too :)